### PR TITLE
Add wrapper for DF driver AK8963

### PR DESCRIPTION
**Pull request by eyeam3**
Add the DF wrapper for the AK8963 driver used on the Parrot Bebop. It also includes some adaptations to use the sensor calibration and the attitude estimation on the Bebop.

However, during testing there have been some problems with the precision of the compass calibration, resulting in a drifting yaw estimation. This needs some further improvements.

### CODE REVIEW

File: src/modules/commander/mag_calibration.cpp [hunk 1]
```diff
@@ -664,7 +664,11 @@ calibrate_return mag_calibrate_all(orb_advert_t *mavlink_log_pub)
 		for (unsigned cur_mag=0; cur_mag<max_mags; cur_mag++) {
 			if (device_ids[cur_mag] != 0) {
 				struct mag_calibration_s mscale;
-#if !defined(__PX4_QURT) && !defined(__PX4_POSIX_RPI)
+				mscale.x_scale = 1.0;
+				mscale.y_scale = 1.0;
+				mscale.z_scale = 1.0;
```
Comment (julianoes) [inline src/modules/commander/mag_calibration.cpp]: That was a good catch!

**Comment by julianoes** [comment]
Reviewed, looks good!

**Comment by LorenzMeier** [comment]
Rebased and applied.

**Comment by EwoudSmeur** [comment]
I wonder what problems you have with the magnetometer calibration... I work at the MAVLab in Delft with the Paparazzi autopilot, and what I noticed on the Bebop2 is that the sensor is in some way nonlinear. 

What I mean is that I do a dance with the Bebop2 and take the maximum and minimum measured values on each axis to do the calibration, such that the scaled values are between 1 and -1. If I then put the drone towards north, I get a value on the X axis of ~0.5, which should be ~0.35 in Delft. If I then turn the drone 180 degrees, I expect the same value, but negative. Instead it is ~ -0.2. If I mess with the offset I do get symmetric values, but then the min and max are no longer plus and minus 1. However, I am able to fly with this changed calibration, whereas I get a spiral otherwise due to yaw estimation errors.

Perhaps the sensor is slightly rotated, or has a nonlinear measurement? I never had problems with the Bebop1 though.

**Comment by eyeam3** [comment]
Yes, that is the exact same problem I encounter. It occurs also on the X axis with roughly the same values you described. Though I use the magnetometer calibration of the PX4 firmware, which performs a least square fit for a sphere. The computed offsets produce an asymmetric signal on the X axis. Similar to you, I adjusted the offsets by hand, resulting in a drift free yaw estimation.

Nevertheless, this is an issue that should be revised. Maybe, the problem could be solved with a more elaborate calibration which gives some scaling factors in addition to the offsets.

**Comment by EwoudSmeur** [comment]
Turns out the magnetometer is located on the gps board, which is angled wrt. the IMU. Thanks for your view on the problem!

**Comment by LorenzMeier** [comment]
@eyeam3 We can set the rotation and probably should do that. Can you plan for that and ping me or @julianoes for details.

**Comment by julianoes** [comment]
Thanks for the note @EwoudSmeur. I've created an issue to track this: #5571.

**Comment by LorenzMeier** [comment]
Yes. You need to look at the physical device

**Pull request closed without merging**